### PR TITLE
Add more debug logging to tiered compilation manager

### PR DESCRIPTION
**Pull request by noahfalk**

### CODE REVIEW

**Comment by noahfalk** [comment]
@davmason - please take a look

cc @dotnet/dotnet-diag @janvorli

File: src/vm/tieredcompilation.cpp [hunk 1]
```diff
@@ -413,6 +425,10 @@ void TieredCompilationManager::ActivateCodeVersion(NativeCodeVersion nativeCodeV
         {
             CodeVersionManager::TableLockHolder lock(pCodeVersionManager);
             hr = ilParent.SetActiveNativeCodeVersion(nativeCodeVersion, TRUE);
+            LOG((LF_TIEREDCOMPILATION, LL_INFO10000, "TieredCompilationManager::ActivateCodeVersion Method=0x%x (%s::%s), code version id=0x%x. [Suspended] SetActiveNativeCodeVersion ret=0x%x\n",
```
Comment (janvorli) [inline src/vm/tieredcompilation.cpp]: The format character for the pMethodDesc is incorrect (%x, should be %p and probably formatted as %pM like in the preexisting log call below)

File: src/vm/tieredcompilation.cpp [hunk 2]
```diff
@@ -400,6 +408,10 @@ void TieredCompilationManager::ActivateCodeVersion(NativeCodeVersion nativeCodeV
         CodeVersionManager::TableLockHolder lock(pCodeVersionManager);
         ilParent = nativeCodeVersion.GetILCodeVersion();
         hr = ilParent.SetActiveNativeCodeVersion(nativeCodeVersion, FALSE);
+        LOG((LF_TIEREDCOMPILATION, LL_INFO10000, "TieredCompilationManager::ActivateCodeVersion Method=0x%x (%s::%s), code version id=0x%x. SetActiveNativeCodeVersion ret=0x%x\n",
```
Comment (janvorli) [inline src/vm/tieredcompilation.cpp]: The format character for the pMethodDesc is incorrect (%x, should be %p and probably formatted as %pM like in the preexisting log call below)

File: src/vm/tieredcompilation.cpp [hunk 3]
```diff
@@ -201,6 +201,10 @@ void TieredCompilationManager::AsyncPromoteMethodToTier1(MethodDesc* pMethodDesc
         }
     }
 
+    LOG((LF_TIEREDCOMPILATION, LL_INFO10000, "TieredCompilationManager::AsyncPromoteMethodToTier1 Method=0x%x (%s::%s), code version id=0x%x queued\n",
```
Comment (janvorli) [inline src/vm/tieredcompilation.cpp]: The format character for the pMethodDesc is incorrect (%x, should be %p and probably formatted as %pM like in the preexisting log call below)

File: src/vm/tieredcompilation.cpp [hunk 4]
```diff
@@ -367,11 +371,15 @@ BOOL TieredCompilationManager::CompileCodeVersion(NativeCodeVersion nativeCodeVe
     EX_TRY
     {
         pCode = pMethod->PrepareCode(nativeCodeVersion);
+        LOG((LF_TIEREDCOMPILATION, LL_INFO10000, "TieredCompilationManager::CompileCodeVersion Method=0x%x (%s::%s), code version id=0x%x, code ptr=0x%x\n",
```
Comment (janvorli) [inline src/vm/tieredcompilation.cpp]: The format character for the pMethodDesc is incorrect (%x, should be %p and probably formatted as %pM like in the preexisting log call below)

File: src/vm/tieredcompilation.cpp (hunk 1 above)
Comment (noahfalk) [reply src/vm/tieredcompilation.cpp]: Ah thanks, I had spotted that elsewhere but forgot to come back. I suspect you'll find the same thing in some of the other recent PRs that I'll need to fix up.

File: src/vm/tieredcompilation.cpp [hunk 5]
```diff
@@ -201,6 +201,10 @@ void TieredCompilationManager::AsyncPromoteMethodToTier1(MethodDesc* pMethodDesc
         }
```
Comment (davmason) [inline src/vm/tieredcompilation.cpp]: Would it be helpful to add logging to any of the failure cases above, where we fail due to it already being optimized, OOM, etc?

Comment (noahfalk) [reply src/vm/tieredcompilation.cpp]: I was mostly just adding logging specific to what I needed to debug, but I went ahead and added those too. I also noticed one of the existing log statements wasn't quite in the right spot so I moved it while I was in there.

**Comment by noahfalk** [comment]
@dotnet-bot test osx10.12 x64 checked build and test

**Comment by noahfalk** [comment]
@dotnet-bot test windows x86 full_opt legacy_backend CoreCLR Perf Tests

**Pull request merged**